stdlib/map.ml

```diff
@@ -312,11 +313,19 @@ module Make(Ord: OrderedType) = struct
           let r' = mapi f r in
           Node{l=l'; v; d=d'; r=r'; h}
 
-    let rec fold f m accu =
+    let rec fold_ascending f m accu =
       match m with
         Empty -> accu
       | Node {l; v; d; r} ->
-          fold f r (f v d (fold f l accu))
+          fold_ascending f r (f v d (fold_ascending f l accu))
```

Comment:
Nitpick: I think this definition (and the _descending one) would be clearer with let bindings for the intermediate accumulator states (I would use three let-bindings and just return the variable bound last). The current code makes it non-obvious to reason about evaluation order (although the two versions are in fact strictly identical).